### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by lukasjuhrich**
__Please don't merge yet!__

These are random improvements I made without a lot of order.

I want to start a discussion which changes are good, and which are disliked (probably the line wrappings). Please comment on the according file changes below.

I tried to add docstrings where things were unclear for me or use a more „pythonic“ way.

I hope the docstring format is correct, since I do not know anything about sphinx or similiar.

I probably will continue to push commits onto this pull request.

### CODE REVIEW

File: mwclient/client.py [hunk 1]
```diff
@@ -121,7 +127,7 @@ def __init__(self, host, path='/w/', ext='.php', pool=None, retry_timeout=30,
                     raise errors.OAuthAuthorizationError(e.code, e.info)
 
                 # Private wiki, do init after login
-                if e.args[0] not in (u'unknown_action', u'readapidenied'):
+                if e.args[0] not in {u'unknown_action', u'readapidenied'}:
```
Comment (lukasjuhrich) [inline mwclient/client.py]: __Ew__!

Turns out 2.6 doesn't know the dict literal. Workaround would be using `set(…)`

Are there any plans on dropping 2.6 support?

**Comment by lukasjuhrich** [comment]
As mentioned in the inline comment, I would hereby ask whether it would be possible to drop 2.6 support.

Python 2.6 has been released 8 years ago and been officially unsupported since [October 29, 2013](https://www.python.org/download/releases/2.6.9/).

**Comment by danmichaelo** [comment]
Sorry I haven't found time to review the commits, but it looks like lots of good stuff 👍 

I'm open for removing Python 2.6 support. Even conservative linux distros like RHEL has had Python 2.7 for some time now. But would you mind creating a separate issue for that? That way more users might see it.

**Comment by lukasjuhrich** [comment]
@danmichaelo will a separate issue be enough, or do you want this in a separate branch/pr as well? Issue will be created.

**Comment by lukasjuhrich** [comment]
@danmichaelo you might want to take a look at both Pull requests #129 (on which this one bases) and #134 and the corresponding issues. Although it's nothing new.

**Comment by danmichaelo** [comment]
I merged this now (with a little bit of rebasing), thanks for writing clear commit messages! I'll try to respond faster in the future if you have more stuff :)

**Comment by lukasjuhrich** [comment]
No problem, thanks a lot for merging and the welcoming tone! :+1:

**Pull request closed without merging**